horreum-web/src/Banner.tsx

```diff
@@ -3,19 +3,58 @@ import { useEffect, useState } from "react"
 import { Alert } from "@patternfly/react-core"
 import {Banner as BannerData, bannerApi} from "./api"
 
+function getAlertBanner(banner: BannerData) {
+    return (
+        <Alert variant={banner.severity as any} title={banner.title} isInline>
+            <div dangerouslySetInnerHTML={{ __html: banner.message || "" }}></div>
+        </Alert>
+    )
+}
+
+// 30 seconds
+const DEFAULT_TIMEOUT = 30000
+export type TimeoutBannerProps = {
+    bannerData: BannerData
+    timeout?: number
+    onTimeout?: () => void
+}
+
+export function TimeoutBanner({bannerData, timeout, onTimeout}: TimeoutBannerProps) {
+    timeout = timeout ?? DEFAULT_TIMEOUT
+    const [banner, setBanner] = useState<BannerData | undefined>(bannerData)
+
+    useEffect(() => {
+        if (banner) {
+            const timeoutId = setTimeout(() => {
+                setBanner(undefined)
+                if (onTimeout) {
+                    onTimeout()
+                }
+            }, timeout)
+
+            return () => clearTimeout(timeoutId);
+        }
+    }, [])
+
+    if (!banner) {
+        return null
+    }
+
+    return getAlertBanner(banner)
+}
+
 export default function Banner() {
     const [banner, setBanner] = useState<BannerData>()
     const [updateCounter, setUpdateCounter] = useState(0)
     useEffect(() => {
-        setTimeout(() => setUpdateCounter(updateCounter + 1), 60000)
+        const timeoutId = setTimeout(() => setUpdateCounter(updateCounter + 1), 60000)
         bannerApi.get().then(setBanner)
+
+        return () => clearTimeout(timeoutId)
     }, [updateCounter])
```

Comment:
I know this isn't a line you changed but I believe we need to include `setUpdateCounter` in the dependency array for this `useEffect` because it is entirely possible for React to regenerate that function and our code to left with a pointer to a function that no longer actually sets the update counter.

Reply:
I am not that sure about that as the `setUpdateCounter` is generated by React's `useState` which should be stable.

I found this [comment](https://stackoverflow.com/questions/71942624/should-a-setstate-function-be-a-dependency-of-useeffect-when-passed-via-hook#:~:text=You%20know%20that%20(because%20the%20setter%20is%20guaranteed%20to%20be%20stable%20by%20useState%2C%20and%20you%27re%20passing%20it%20verbatim%20through%20context%20and%20your%20useMyState%20hook)) on Stackoverflow mentioning it.